Blog post for sprint 107 about Workflow Runs (#329)
openSUSE/obs-landing: _posts/development/2021-11-22-scm-workflow-runs.md

---
layout: post
title: SCM/CI Workflow Runs and More
category: development
---

> We go further with the SCM/CI workflow integration in OBS. You, beta testers, have had difficulties learning why your integration failed so far. This is solved with the new  _Workflow Runs_ UI feature, with detailed information about every workflow that runs once you trigger a workflow token. Keep reading for more details.

Reviewer: ```suggestion
We go further with the SCM/CI workflow integration in OBS. You, beta testers, had difficulties in understanding why your integration failed when something doesn't go as planned. This is solved with the new _workflow runs_ UI feature, with detailed information about every workflow that ran once you triggered a workflow token. Keep reading for more details.
```

Author: I rephrased it a bit in the past tense.

Do not forget this feature is under the [the beta program](/2018/10/04/the-beta-program/). Join!

{% include partials/_series-of-posts-about-scm-integration.md %}

# Tokens Web UI

At this point, you might be familiar with the tokens web UI. After a few improvements, it looks like follows.

<figure>

>   <img src="/images/posts/sprint_107_tokens_ui_index.png" alt="Lastest version of the token web UI" />

Reviewer: ```suggestion
  <img src="/images/posts/sprint_107_tokens_ui_index.png" alt="Latest version of the tokens web UI" />
```

>   <figcaption>Lastest version of the token web UI</figcaption>

Reviewer: ```suggestion
  <figcaption>Latest version of the token web UI</figcaption>
```

</figure>

> Now the token id is visible from the list of tokens and we have introduced a new link to workflow runs <img src="/images/posts/sprint_107_workflow_runs_icon.png" alt="Workflow runs icon" />. 

Reviewer: ```suggestion
The token id is now visible from the list of tokens and we have introduced a new link to the workflow runs <img src="/images/posts/sprint_107_workflow_runs_icon.png" alt="Workflow runs icon" />. 
```

From now on, every time a workflow token is triggered, OBS will track information about each workflow run.

<figure>
  <img src="/images/posts/sprint_107_tokens_ui_index_with_arrow.png" alt="Workflow runs link" />
  <figcaption>Workflow runs link</figcaption>
</figure>

You will get details like:

- the start time and the status of the workflow run (running/fail/success) represented by icons;
<figure>
  <img src="/images/posts/sprint_107_workflow_runs_list.png" alt="Workflow runs list" />
  <figcaption>Workflow runs list</figcaption>
</figure>

- the request headers and the payload received from the SCM;

<figure>
  <img src="/images/posts/sprint_107_workflow_runs_request.png" alt="Workflow runs request tab" />
  <figcaption>Workflow runs request tab</figcaption>
</figure>

> - the response body sent from OBS to the SCM, containing the error message in case of a failed run.

Reviewer: ```suggestion
- the response body sent from OBS to the SCM, containing error messages in case of a failed run.
```

<figure>
  <img src="/images/posts/sprint_107_workflow_runs_response.png" alt="Workflow runs response tab" />
  <figcaption>Workflow runs response tab</figcaption>
</figure>

Although this tool will be of great help to you, we recommend that you follow the [instructions](https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_ci_workflow_integration.html) thoroughly for successful integration.

# And Much More

> Apart from the big changes in the web UI, we continue improving other areas. We now handle other possible errors and display a more readable version of the text in the SCM checks by using the short version of the commit sha.

Reviewer: ```suggestion
Apart from the big changes in the web UI, we continue improving other areas. We now handle other possible errors and display a more readable version of the text in the SCM checks by using the short version of commit SHAs for push events.
```

<figure>

>   <img src="/images/posts/sprint_107_short_commit_sha.png" alt="SCM checks display short commit sha" />

Reviewer: ```suggestion
  <img src="/images/posts/sprint_107_short_commit_sha.png" alt="SCM checks display short commit SHA" />
```

>   <figcaption>SCM checks display short commit sha</figcaption>

Reviewer: ```suggestion
  <figcaption>SCM checks display short commit SHA</figcaption>
```

  <figcaption>Workflow runs link</figcaption>
</figure>

You will get details like:

- the start time and the status of the workflow run (running/fail/success) represented by icons;
<figure>
  <img src="/images/posts/sprint_107_workflow_runs_list.png" alt="Workflow runs list" />
  <figcaption>Workflow runs list</figcaption>
</figure>

- the request headers and the payload received from the SCM;

<figure>
  <img src="/images/posts/sprint_107_workflow_runs_request.png" alt="Workflow runs request tab" />
  <figcaption>Workflow runs request tab</figcaption>
</figure>

- the response body sent from OBS to the SCM, containing the error messages in case of a failed run.

<figure>
  <img src="/images/posts/sprint_107_workflow_runs_response.png" alt="Workflow runs response tab" />
  <figcaption>Workflow runs response tab</figcaption>
</figure>

Although this tool will be of great help to you, we recommend that you follow the [instructions](https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_ci_workflow_integration.html) thoroughly for successful integration.

# Push Events for Tags

> Our workflows now accept push events for tags. You could, for example, run a certain workflow only when a tag is pushed to a repository rather than on every push to a branch.

Author: Ok!

> Our workflows now accept push events for tags. You could, for example, run a certain workflow only when a tag is pushed to a repository rather than on every push to a branch.

Author: Changes pushed, please have a look.

> These kinds of events will affect the branch package and the link package steps only. Read the [documentation about the steps](https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_ci_workflow_integration.html) for more details.

Reviewer 2: ```suggestion
These kinds of events will affect the branch package and the link package steps only. Read the [documentation about the steps](https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_ci_workflow_integration.html#sec.obs.obs_scm_ci_workflow_integration.setup.obs_workflows.steps) for more details.
```

We can be more precise adding the specific section of the book.

Author: true!

Author: change pushed, please have a look
